pkg/sdkserver/sdkserver.go

```diff
@@ -978,48 +992,258 @@ func (s *SDKServer) updateCounter(ctx context.Context) error {
+// RemoveListValue collapses all remove a value from a List requests into a single UpdateList request.
+// Returns not found if the List does not exist.
+// Returns not found if the value is not in the List.
 // [Stage:Alpha]
 // [FeatureFlag:CountsAndLists]
 func (s *SDKServer) RemoveListValue(ctx context.Context, in *alpha.RemoveListValueRequest) (*alpha.List, error) {
 	if !runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
 		return nil, errors.Errorf("%s not enabled", runtime.FeatureCountsAndLists)
 	}
-	// TODO(#2716): Implement Me
-	return nil, errors.Errorf("Unimplemented -- RemoveListValue coming soon")
+
+	s.logger.WithField("name", in.Name).Debug("Remove List Value")
+
+	gs, err := s.gameServer()
+	if err != nil {
+		return nil, err
+	}
+
+	s.gsUpdateMutex.Lock()
+	defer s.gsUpdateMutex.Unlock()
+
+	if list, ok := gs.Status.Lists[in.Name]; ok {
+		// Verify value exists in the list
+		for _, val := range list.Values {
+			if in.Value != val {
+				continue
+			}
+			// Add value to remove to gsListUpdates map.
+			batchList := s.gsListUpdates[in.Name]
+			if batchList.valuesToDelete == nil {
+				batchList.valuesToDelete = map[string]bool{}
+			}
+			batchList.valuesToDelete[in.Value] = true
+			s.gsListUpdates[in.Name] = batchList
+			// Queue up the Update for later batch processing by updateLists.
+			s.workerqueue.Enqueue(cache.ExplicitKey(updateLists))
+			return &alpha.List{}, nil
+		}
+		return nil, errors.Errorf("not found. Value: %s not found in List: %s", in.Value, in.Name)
+	}
+	return nil, errors.Errorf("not found. %s List not found", in.Name)
+}
+
+// updateList updates the Lists in the GameServer's Status with the batched update list requests.
+// Includes all SetCapacity, AddValue, and RemoveValue requests in the batched request.
+func (s *SDKServer) updateList(ctx context.Context) error {
+	gs, err := s.gameServer()
+	if err != nil {
+		return err
+	}
+	gsCopy := gs.DeepCopy()
+
+	s.logger.WithField("batchListUpdates", s.gsListUpdates).Debug("Batch updating List(s)")
+	s.gsUpdateMutex.Lock()
+	defer s.gsUpdateMutex.Unlock()
+
+	names := []string{}
+
+	for name, listReq := range s.gsListUpdates {
+		list, ok := gsCopy.Status.Lists[name]
+		if !ok {
+			continue
+		}
+		if listReq.capacitySet != nil {
+			list.Capacity = *listReq.capacitySet
+		}
+		if len(listReq.valuesToDelete) != 0 {
+			list.Values = deleteValues(list.Values, listReq.valuesToDelete)
+		}
+		if len(listReq.valuesToAppend) != 0 {
+			list.Values = agonesv1.MergeRemoveDuplicates(list.Values, listReq.valuesToAppend)
+		}
+
+		if int64(len(list.Values)) > list.Capacity {
+			s.logger.Debugf("truncating Values in Update List request to List Capacity %d", list.Capacity)
+			list.Values = append([]string{}, list.Values[:list.Capacity]...)
+		}
+		gsCopy.Status.Lists[name] = list
+		names = append(names, name)
+	}
+
+	gs, err = s.gameServerGetter.GameServers(s.namespace).Update(ctx, gsCopy, metav1.UpdateOptions{})
+	if err != nil {
+		return err
```

Comment:
Should we add a log line here? `s.recorder.Event(gs, corev1.EventTypeWarning, "UpdateList", fmt.Sprintf("Unable to update List(s) with batched change requests: %v", s.gsListUpdates))` Or will this automatically retry? The changes aren't lost -- they're in s.gsCounterUpdates, but will it get added back to the queue?

Reply:
Any return of an error means the worker queue will retry (all our workerqueues are like this), so you don't have to do anything extra, and I don't think a log is required. It's expected to fail when out of sync.